### PR TITLE
Loadtestservice - SDK initial review

### PR DESCRIPTION
**Pull request by NiveditJain**
Continuing thread from: https://github.com/Azure/azure-sdk-for-python/pull/24744 

Regenerated using the command, 

```
autorest --version=3.8.1 --use=@autorest/python@6.1.0 --use=@autorest/modelerfour@4.23.5 README.md
```

### CODE REVIEW

**Comment by NiveditJain** [comment]
API View: https://apiview.dev/Assemblies/Review/85d28ad8d2144aeba248f2886d136a31

File: sdk/loadtestservice/azure-developer-loadtest/README.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+
+# Azure Developer LoadTesting client library for Python
+<!-- write necessary description of service -->
```
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtest/README.md:3]: Please write some overview about the service to override these words

Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtest/README.md]: Added in latest commit

File: .vscode/cspell.json [hunk 2]
```diff
@@ -461,6 +461,12 @@
         "undoc"
       ]
     },
+    {
+      "filename": "sdk/loadtestservice/azure-developer-loadtest/azure/developer/loadtesting/**",
+      "words":[
+        "vusers"
+      ]
+    },
```
Comment (msyyc) [inline .vscode/cspell.json:469]: There seems be many words which fail cspell check: https://dev.azure.com/azure-sdk/public/_build/results?buildId=1720898&view=logs&jobId=b70e5e73-bbb6-5567-0939-8415943fadb9&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=bdeefc16-b669-5ebd-ad94-a2c19ade53b0. it is more convenient to add your folder here: https://github.com/Azure/azure-sdk-for-python/blob/37abf4e4da05bd776226faeb923cf3699f4548d2/.vscode/cspell.json#L12

File: sdk/loadtestservice/azure-developer-loadtest/tests/test_smoke.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        client = self.create_client(loadtestservice_endpoint)
+        result = client.administration.test.list_load_test_search()
+        assert result is not None
+
+    @LoadtestservicePowerShellPreparer()
+    def test_smoke_delete_test(self, loadtestservice_endpoint):
+        client = self.create_client(loadtestservice_endpoint)
+        test_id = "000001"
+        result = client.administration.test.delete_load_test(test_id)
+        assert result is None
```
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtest/tests/test_smoke.py:42]: I find that you customize some function(like: https://github.com/Azure/azure-sdk-for-python/pull/25264/files#r925285754). It is better to test it for better code quality.

File: sdk/loadtestservice/azure-developer-loadtest/samples/sample.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+# creating app component
+try:
+    result = client.load_test_administration.create_or_update_app_components(
+        APP_COMPONENT,
+        {
+            "name": "app_component",
+            "testId": TEST_ID,
+            "value": {
+                "/subscriptions/7c71b563-0dc0-4bc0-bcf6-06f8f0516c7a/resourceGroups/App-Service-Sample-Demo-rg/providers/Microsoft.Web/sites/App-Service-Sample-Demo": {
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtest/samples/sample.py]: Resolved in: https://github.com/Azure/azure-sdk-for-python/pull/25264/commits/9ffe28b96e30306aca8d6fffc6f90d70c7f36733

File: sdk/loadtestservice/azure-developer-loadtest/README.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+### Installating the package
+
+```bash
+python -m pip install azure-developer-loadtesting
+```
+
+#### Prequisites
+
+- Python 3.6 or later is required to use this package.
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtest/README.md]: Resolved

File: sdk/loadtestservice/azure-developer-loadtest/README.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+## Examples
+
+```python
+>>> from azure.developer.loadtesting import LoadTestClient
+>>> from azure.identity import DefaultAzureCredential
+>>> from azure.core.exceptions import HttpResponseError
+
+>>> client = LoadTestClient(endpoint='<endpoint>', credential=DefaultAzureCredential())
+>>> try:
+        <!-- write test code here -->
```
[the remark this answers is not in the document]
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtest/README.md]: resolved in https://github.com/Azure/azure-sdk-for-python/pull/25264/commits/568cc6570164db6cf4534b63b1db164034d6c0b7

**Comment by NiveditJain** [comment]
Hi @iscai-msft, 
I have resolved your comments on https://github.com/Azure/azure-sdk-for-python/pull/24744 here, kindly review and let me know your feedback. 

Thanks!

File: sdk/loadtestservice/azure-developer-loadtest/tests/test_smoke.py (hunk 3 above)
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtest/tests/test_smoke.py]: resolved in https://github.com/Azure/azure-sdk-for-python/pull/25264/commits/e1e1153068ebd6d08c01c23194cd338add7857f6

File: sdk/loadtestservice/azure-developer-loadtest/tests/test_smoke_async.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        #         "secrets": {},
+        #         "environmentVariables": {},
+        #         "passFailCriteria": {"passFailMetrics": {}},
+        #         "keyvaultReferenceIdentityType": "SystemAssigned",
+        #         "keyvaultReferenceIdentityId": None,
+        #     }
+        # )
+        # test your code here, for example:
+        # result = await client.xxx.xx(...)
+        # assert result is not None
```
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtest/tests/test_smoke_async.py:47]: why is it removed?

File: sdk/loadtestservice/azure-developer-loadtest/setup.py [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        "isodate>=0.6.0",
+        
+        
+        "azure-core<2.0.0,>=1.2.2",
+        
+    ],
+    
+    python_requires=">=3.6",
+    
+)
```
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtest/setup.py:86]: The code format is in a mess. Please update code with `autorest --version=3.8.1 --use=@autorest/python@6.0.0 --use=@autorest/modelerfour@4.23.5 README.md`

File: shared_requirements.txt [hunk 9]
```diff
@@ -225,6 +225,7 @@ opentelemetry-sdk<2.0.0,>=1.5.0,!=1.10a0
 #override azure-mixedreality-remoterendering azure-core<2.0.0,>=1.6.0
 #override azure-mixedreality-remoterendering azure-mixedreality-authentication>=1.0.0b1
 #override azure-ai-language-conversations azure-core<2.0.0,>=1.24.0
+#override azure-analytics-loadtestservice azure-core<2.0.0,>=1.23.0
```
Comment (msyyc) [inline shared_requirements.txt:228]: You could reference https://github.com/Azure/azure-sdk-for-python/wiki/How-to-fix-CI#problems-and-resolution to fix ci

File: sdk/loadtestservice/azure-developer-loadtest/README.md [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+from azure.developer.loadtesting import LoadTestClient
+from azure.identity import DefaultAzureCredential
+from azure.core.exceptions import HttpResponseError
+
+client = LoadTestClient(endpoint='<endpoint>', credential=DefaultAzureCredential())
+try:
+    result = client.load_test_administration.create_or_update_test(
+        TEST_ID,
+        {
+            "resourceId": f"/subscriptions/{SUBSCRIPTION_ID}/resourceGroups/yashika-rg/providers/Microsoft.LoadTestService/loadtests/loadtestsdk",
```
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtest/README.md:53]: I guess you mean `SUBSCRIPTION_ID` is from environment variables but I think it is better to declare it before use it like:
`SUBSCRIPTION_ID = os.getenv("SUBSCRIPTION_ID ")`

File: sdk/loadtestservice/azure-developer-loadtest/README.md [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+            "displayName": DISPLAY_NAME,
+            "requestSamplers": [],
+            "errors": [],
+            "percentiles": ["90"],
+            "groupByInterval": "5s",
+        },
+    )
+    print(result)
+except HttpResponseError as e:
+    print("Failed to send JSON message: {}".format(e.response.json()))
```
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtest/README.md:103]: it is same for other variables (`TEST_RUN_ID`, `TEST_ID`, etc): https://github.com/Azure/azure-sdk-for-python/pull/25264/files#r927257552

File: sdk/loadtestservice/azure-developer-loadtest/setup.py (hunk 8 above)
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtest/setup.py]: Cleaned setup.py in https://github.com/Azure/azure-sdk-for-python/pull/25264/commits/4a6d627c9abe708dd37ff12abbcd6ec8dbcbd20b

File: sdk/loadtestservice/azure-developer-loadtest/README.md (hunk 11 above)
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtest/README.md]: Resolved in https://github.com/Azure/azure-sdk-for-python/pull/25264/commits/faddc05c876a7926ac822e27fce970217dab4929

File: sdk/loadtestservice/azure-developer-loadtest/README.md (hunk 10 above)
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtest/README.md]: Resolved in https://github.com/Azure/azure-sdk-for-python/pull/25264/commits/faddc05c876a7926ac822e27fce970217dab4929

File: sdk/loadtestservice/azure-mgmt-loadtestservice/azure/mgmt/loadtestservice/dev_requirements.txt [hunk 12]
```diff
@@ -0,0 +1,5 @@
+-e ../../../tools/azure-devtools
+-e ../../../tools/azure-sdk-tools
+../../core/azure-core
+../../identity/azure-identity
+aiohttp
```
Comment (msyyc) [inline sdk/loadtestservice/azure-mgmt-loadtestservice/azure/mgmt/loadtestservice/dev_requirements.txt:5]: Why do you change `azure-mgmt-loadtestservice`? It is another folder

File: sdk/loadtestservice/azure-mgmt-loadtestservice/dev_requirements.txt [hunk 13]
```diff
@@ -0,0 +1,2 @@
+-e ../../../tools/azure-devtools
+-e ../../../tools/azure-sdk-tools
```
Comment (msyyc) [inline sdk/loadtestservice/azure-mgmt-loadtestservice/dev_requirements.txt:2]: same with https://github.com/Azure/azure-sdk-for-python/pull/25264/files#r928362974

File: sdk/loadtestservice/ci.yml [hunk 14]
```diff
@@ -29,5 +29,5 @@ extends:
   parameters:
     ServiceDirectory: loadtestservice
     Artifacts:
-    - name: azure-mgmt-loadtestservice
-      safeName: azuremgmtloadtestservice
+    - name: azure-developer-loadtesting
+      safeName: azuredeveloperloadtesting
```
Comment (msyyc) [inline sdk/loadtestservice/ci.yml:33]: you could add new line but you can't change the existing lines which has been published

File: sdk/loadtestservice/azure-developer-loadtest/setup.py (hunk 8 above)
Comment (msyyc) [reply sdk/loadtestservice/azure-developer-loadtest/setup.py]: `autorest --version=3.8.1 --use=@autorest/python@6.0.0 --use=@autorest/modelerfour@4.23.5 README.md` this cmd shall help to reformat the generated sdk code. There are many blank lines in `setup.py`: https://github.com/NiveditJain/azure-sdk-for-python/blob/f3039cb36e24adfa67710893c7f0b6564bd9a279/sdk/loadtestservice/azure-developer-loadtesting/setup.py#L78-L79

File: sdk/loadtestservice/azure-developer-loadtesting/README.md [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+## Examples
+
+### Creating a load test 
+```python
+from azure.developer.loadtesting import LoadTestClient
+from azure.identity import DefaultAzureCredential
+from azure.core.exceptions import HttpResponseError
+
+TEST_ID = "a011890b-0201-004d-010d"  
+DISPLAY_NAME = "new_namespace-new-namespace"
```
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:49]: Is it secure to expose these id?

File: sdk/loadtestservice/azure-developer-loadtesting/tests/test_smoke_async.py [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+        #             "splitAllCSVs": False,
+        #         },
+        #         "secrets": {},
+        #         "environmentVariables": {},
+        #         "passFailCriteria": {"passFailMetrics": {}},
+        #         "keyvaultReferenceIdentityType": "SystemAssigned",
+        #         "keyvaultReferenceIdentityId": None,
+        #     }
+        # )
+        # assert result is not None
```
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtesting/tests/test_smoke_async.py:44]: Why is the async testcase annotated?

File: sdk/loadtestservice/azure-developer-loadtesting/setup.py [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+    author="Microsoft Corporation",
+    author_email="azpysdkhelp@microsoft.com",
+    url="https://github.com/Azure/azure-sdk-for-python/tree/main/sdk",
+    keywords="azure, azure sdk",
+    classifiers=[
+        "Development Status :: 4 - Beta",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3 :: Only",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.6",
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/setup.py:45]: Remove 3.6

File: sdk/loadtestservice/azure-developer-loadtesting/setup.py [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+    ),
+    include_package_data=True,
+    package_data={
+        'pytyped': ['py.typed'],
+    },
+    install_requires=[
+        "isodate<1.0.0,>=0.6.1",
+        "azure-core<2.0.0,>=1.24.0",
+    ],
+    python_requires=">=3.6",
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/setup.py:69]:
```suggestion
    python_requires=">=3.7",
```

File: sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md [hunk 19]
```diff
@@ -0,0 +1,5 @@
+# Release History
+
+## 1.0.0b1 (1970-01-01)
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md:3]: Be sure to put the right date

File: sdk/loadtestservice/azure-developer-loadtesting/setup.py (hunk 17 above)
Comment (msyyc) [reply sdk/loadtestservice/azure-developer-loadtesting/setup.py:45]: Just need to update the code with `autorest --version=3.8.1 --use=@autorest/python@6.1.0 --use=@autorest/modelerfour@4.23.5 README.md` and the template will be updated automatically

File: sdk/loadtestservice/azure-developer-loadtesting/setup.py (hunk 18 above)
Comment (msyyc) [reply sdk/loadtestservice/azure-developer-loadtesting/setup.py:69]: same with https://github.com/Azure/azure-sdk-for-python/pull/25264#discussion_r930556621

**Review by msyyc — APPROVED** [review]

File: sdk/loadtestservice/azure-developer-loadtesting/README.md [hunk 20]
```diff
@@ -0,0 +1,180 @@
...
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables:
+`AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`
+
+Use the returned token credential to authenticate the client:
+
+```python
+from azure.developer.loadtesting import LoadTestClient
+from azure.identity import DefaultAzureCredential
+client = LoadTestClient(endpoint='<endpoint>', credential=DefaultAzureCredential())
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:37]: Should be LoadTestingClient

File: sdk/loadtestservice/azure-developer-loadtesting/README.md [hunk 21]
```diff
@@ -0,0 +1,180 @@
...
+```python
+from azure.developer.loadtesting import LoadTestClient
+from azure.identity import DefaultAzureCredential
+from azure.core.exceptions import HttpResponseError
+
+TEST_ID = "some-test-id"  
+DISPLAY_NAME = "new_namespace-new-namespace"  
+SUBSCRIPTION_ID = os.environ["SUBSCRIPTION_ID"]  
+
+client = LoadTestClient(endpoint='<endpoint>', credential=DefaultAzureCredential())
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:52]: Should be LoadTestingClient

File: sdk/loadtestservice/azure-developer-loadtesting/README.md [hunk 22]
```diff
@@ -0,0 +1,180 @@
...
+
+    result = client.load_test_administration.upload_test_file(TEST_ID, FILE_ID, body)
+    print(result)
+except HttpResponseError as e:
+    print("Failed to send JSON message: {}".format(e.response.json()))
+```
+
+### Running a Test
+```python
+from azure.developer.loadtesting import LoadTestClient
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:100]: Should be LoadTestingClient

File: sdk/loadtestservice/azure-developer-loadtesting/README.md [hunk 23]
```diff
@@ -0,0 +1,180 @@
...
+from azure.identity import DefaultAzureCredential
+from azure.core.exceptions import HttpResponseError
+
+TEST_ID = "some-test-id"  
+TEST_RUN_ID = "some-testrun-id" 
+DISPLAY_NAME = "new_namespace-new-namespace"  
+
+try:
+    result = client.load_test_runs.create_and_update_test(
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:109]: Client has not been created in that sample

File: sdk/loadtestservice/azure-developer-loadtesting/README.md [hunk 24]
```diff
@@ -0,0 +1,180 @@
...
+    )
+    print(result)
+except HttpResponseError as e:
+     print('service responds error: {}'.format(e.response.json()))
+
+```
+
+### Uploading .jmx file to a Test
+```python
+from azure.developer.loadtesting import LoadTestClient
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:80]: Should be LoadTestingClient

File: sdk/loadtestservice/azure-developer-loadtesting/README.md [hunk 25]
```diff
@@ -0,0 +1,180 @@
...
+
+TEST_ID = "some-test-id"  
+FILE_ID = "some-file-id"  
+
+try:
+    # opening .jmx file
+    body = {}
+    body["file"] = open("sample.jmx", "rb")
+
+    result = client.load_test_administration.upload_test_file(TEST_ID, FILE_ID, body)
```
Comment (lmazuel) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:92]: client has not been create in that sample

**Pull request merged**